include/deal.II/base/vectorization.h

```diff
@@ -78,6 +79,37 @@ struct EnableIfScalar<VectorizedArray<Number> >
 };
 
 
+
+#ifndef DEAL_II_WITH_CXX11
+// Specify the types for the implemented multiplications explicitly
+
+template <typename Number>
+struct ProductType<Number, VectorizedArray<Number> >
+{
+  typedef VectorizedArray<Number> type;
+};
+
+template <typename Number>
+struct ProductType<VectorizedArray<Number>, Number>
+{
+  typedef VectorizedArray<Number> type;
+};
+
+template<>
+struct ProductType<double, VectorizedArray<float> >
+{
+  typedef VectorizedArray<float> type;
```

Comment:
Let's call it a feature ;-)

The reason why these functions exist is to allow for doing something like `VectorizedArray<float> * 0.5`. Without it, you'd have to write `... * 0.5f` which is not only unexpected but also prevents switching between `double` and `float` execution based on a typedef. I must admit I know too little about the basic types whether there would have been a different solution for typed numbers.

In any case, the conversion in the other direction does definitely not work: A float vectorized array is 8 wide, but a double vectorized array is only 4 wide. If you did the conversion based on the scalar double, I'd have to throw away half of the vector.

Based on these two observations, I think that the vectorized (SIMD) type must take precedence. I agree that things are different for scalar types and it would have been nice to do it the other way around, but clearly different widths do not allow for that.